compiler/rustc_passes/src/loops.rs

```diff
@@ -233,6 +233,12 @@ impl<'a, 'hir> CheckLoopVisitor<'a, 'hir> {
             Normal | AnonConst => {
                 struct_span_err!(self.sess, span, E0268, "`{}` outside of a loop", name)
                     .span_label(span, format!("cannot `{}` outside of a loop", name))
+                    .span_suggestion(
+                        span,
+                        "consider removing the",
+                        name.into(),
+                        rustc_errors::Applicability::MachineApplicable,
+                    )
```

Comment:
```suggestion
                    .span_suggestion(
                        span,
                        &format!("consider removing the `{}`", name),
                        String::new(),
                        Applicability::MaybeIncorrect,
                    )
```

Please add a `use` at the top of the file for `rustc_errors::Applicability` too.

----

In an ideal world we would also look for potential places where the user wanted to iterate, but this is fine for now.